core/src/main/java/org/elasticsearch/gateway/ReplicaShardAllocator.java

```diff
@@ -179,7 +179,7 @@ public boolean allocateUnassigned(RoutingAllocation allocation) {
                 }
             } else if (matchingNodes.hasAnyData() == false) {
                 // if we didn't manage to find *any* data (regardless of matching sizes), check if the allocation of the replica shard needs to be delayed
-                changed |= ignoreUnassignedIfDelayed(unassignedIterator, shard);
+                ignoreUnassignedIfDelayed(unassignedIterator, shard);
```

Comment:
I like that change in particular because it brings us in line with how we handle `unassignedIterator.removeAndIgnore()` in all other cases (we never set changed to true just for ignoring a shard).